### DISCUSSION TITLE
How to provide external kafka ssl certificate path

### DISCUSSION THREAD

**Discussion by lashsingh** [milvus-io/milvus#27486] [Q&A and General discussion]
Hi everyone,

My external Kafka instance does not support sasl mechanism plain with username & password, it supports SSL authentication with certs. Is Kafka SSL authentication supported and is there a way to set SSL certification path for external kafka ?

I can see the similar question asked here over a year ago: https://github.com/milvus-io/milvus/discussions/19672

I was wondering if this is possible now.

Thanks.

**Comment by filip-halt** [comment]
@xiaofan-luan Are we looking to add this feature?

**Comment by LoveEachDay** [comment]
@lashsingh Could you raise a issue so that we may add to roadmap?

**Reply by lashsingh** [reply]
Done.

https://github.com/milvus-io/milvus/issues/27563

**Reply by lashsingh** [reply]
@LoveEachDay @xiaofan-luan @jaime0815 I have a working example of Go client code to make Kafka tls connection with certs written using Sarama library. I can see Milvus currently uses confluent-kafka-go library, so not sure if  my code will be help but happy to share if you like to help expedite this feature.

**Reply by yhmo** [reply]
@lashsingh 
If possible, you can submit a pull request in milvus repo.

**Reply by lashsingh** [reply]
@yhmo Sure, will try. In the meantime, please do update thread if you get to add the working solution before I create a PR. Thanks.

**Reply by yhmo** [reply]
Sure

**Reply by lashsingh** [reply]
@yhmo @LoveEachDay I have the sample code ready to help you add this feature but donot have the access to push my branch. I'm getting the following message:

`ERROR: Permission to milvus-io/milvus.git denied to lashsingh.`

**Reply by yhmo** [reply]
Open-source projects don't allow push commits directly. You should fork the project to your domain.
Steps:
1. click the "Fork" button on the https://github.com/milvus-io/milvus page, fork this project to your repositories
![Screenshot from 2023-11-03 10-03-22](https://github.com/milvus-io/milvus/assets/2282099/e5c05e15-629a-4e0e-b894-64090da00e31)

2. `git clone [your forked project]` ------------------------- download the forked project to your local 
3. `git remote add source git@github.com:milvus-io/milvus.git` ---------------------------- add the original project 
4. `git fetch source` -------------------------------------------- fetch the original project git history to local
5. `git branch kkk source/master` ---------------------------- create a local branch from the original project's master branch
6. make code change
7. `git add [the files changed]`
8. `git commit -sm 'commit description'` ---------------------- make a commit
9. `git push origin kkk` ------------------ upload the local kkk branch to your forked project
10. open your forked project page, click "Contribute" and "Open pull request"
![Screenshot from 2023-11-03 10-04-19](https://github.com/milvus-io/milvus/assets/2282099/58fdff8d-eb01-42f2-9ba5-f2b4faa37341)

11. after the pull request is created, you will see your pr in the milvus pull requests page: https://github.com/milvus-io/milvus/pulls
12. wait the pr approved and merged
